### PR TITLE
feat: add real size to backing image

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/8757

- store `realsize` of backingimage in status
- add the `realsize` to `storageScheduled` in diskStatus
- add new field `scheduledBackingImage` in diskStatus

### CODE REVIEW

**Comment by ChanYiLin** [comment]
Hi @derekbit @shuo-wu  
This PR is ready for review, thanks!

File: controller/backing_image_controller.go [hunk 1]
```diff
@@ -840,6 +840,21 @@ func (bic *BackingImageController) syncBackingImageFileInfo(bi *longhorn.Backing
 				}
 			}
 		}
+		if info.RealSize > 0 {
+			if bi.Status.RealSize == 0 {
+				bi.Status.RealSize = info.RealSize
+				bic.eventRecorder.Eventf(bi, corev1.EventTypeNormal, constant.EventReasonUpdate, "Set realSize to %v", bi.Status.RealSize)
```
[the remark this answers is not in the document]
Comment (shuo-wu) [reply controller/backing_image_controller.go:846]: I am fine with this update. Backing image actual/real size update should be an one-time update.

File: controller/backing_image_controller.go [hunk 2]
```diff
@@ -840,6 +840,21 @@ func (bic *BackingImageController) syncBackingImageFileInfo(bi *longhorn.Backing
...
+			}
+			if bi.Status.RealSize != info.RealSize {
+				if bi.Status.DiskFileStatusMap[bim.Spec.DiskUUID].State != longhorn.BackingImageStateFailed {
+					msg := fmt.Sprintf("found mismatching realSize %v reported by backing image manager %v in disk %v, the realSoze recorded in status is %v",
+						info.RealSize, bim.Name, bim.Spec.DiskUUID, bi.Status.RealSize)
+					log.Error(msg)
```
Comment (shuo-wu) [inline controller/backing_image_controller.go:852]: NIT: Use `bic.eventRecorder`

**Review by shuo-wu — APPROVED** [review]
In general LGTM

File: controller/backing_image_controller.go (hunk 2 above)
Comment (ChanYiLin) [reply controller/backing_image_controller.go:852]: fixed.

File: controller/backing_image_controller.go [hunk 3]
```diff
@@ -840,6 +840,21 @@ func (bic *BackingImageController) syncBackingImageFileInfo(bi *longhorn.Backing
...
 			}
 		}
+		if info.RealSize > 0 {
+			if bi.Status.RealSize == 0 {
+				bi.Status.RealSize = info.RealSize
+				bic.eventRecorder.Eventf(bi, corev1.EventTypeNormal, constant.EventReasonUpdate, "Set realSize to %v", bi.Status.RealSize)
+			}
+			if bi.Status.RealSize != info.RealSize {
+				if bi.Status.DiskFileStatusMap[bim.Spec.DiskUUID].State != longhorn.BackingImageStateFailed {
+					msg := fmt.Sprintf("found mismatching realSize %v reported by backing image manager %v in disk %v, the realSoze recorded in status is %v",
```
[the remark this answers is not in the document]
Comment (ChanYiLin) [reply controller/backing_image_controller.go:850]: fixed

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**